### ISSUE TITLE
Add in deep learning with either scikit-learn's MLP, or TF Learn

### ISSUE THREAD

**Issue by ClimbsRocks** [ClimbsRocks/auto_ml#58]
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/learn/python/learn

**Comment by automaton8** [comment]
First off, thanks for creating this repo. Awesome!
Second, would it be possible to include theano as well?

**Comment by ClimbsRocks** [comment]
1. Thanks for using it! I'd love any feedback you have, or to hear about any issues you run into. I'll probably be releasing the next mini version tonight with a number of small bugfixes and improvements, so hopefully you'll see it getting rapidly better :)

2. I like the idea, but that's probably outside the scope for MVP deep learning implementation. Do you have any specific reason to prefer Theano over TensorFlow, or the Multi-Layer Perceptron (ie, deep neural net) that scikit-learn is building out?

**Comment by automaton8** [comment]
Actually I'm using Keras on top of Theano, so could switch to TF as the backend. I guess then my request is to include Keras support?  Not so keen on sklearn mlp as it doesnt support GPU.

**Comment by ClimbsRocks** [comment]
yeah, totally understandable. i'll see what i can do with tensorflow as soon as i get the chance. @automaton8 would love your thoughts on a good, generalized setup for deep learning that would work well as a default across a range of different problems.

**Comment by ClimbsRocks** [comment]
Just to report some progress here: The basic structure for adding in deep learning is ready! 

We're using TensorFlow, so we should be able to take advantage of GPUs on Linux machines (they only support GPU optimization on Linux for now). 

Now the devil's in the details. Stuff like saving the file is done very differently. And hyperparameter optimization is critically important for deep learning networks, so I want to get a reasonable set of defaults in place for that. 

If anyone wants to jump in and help on any of those sub-tasks, I'd love the assistance!

**Comment by ClimbsRocks** [comment]
@automaton8 the keras branch is nearly ready! if you want to take a look at it, i'd love your thoughts. Closing this issue in favor of a more recent tracking issue covering details related to the current keras branch.
